Doc/c-api/object.rst

```diff
@@ -196,6 +196,17 @@ Object Protocol
       This function now includes a debug assertion to help ensure that it
       does not silently discard an active exception.
 
+
+.. c:function:: PyObject* _PyObject_FunctionStr(PyObject *func)
+
+   Return a user-friendly string representation of the function-like object
+   *func*. This returns ``func.__qualname__ + "()"`` if there is a
```

Comment:
Returning `__qualname__` without module is not particularly useful.

If you want to return a full qualified name (long but unambiguous), return `__module__ + '.' + __qualname__`. `__module__` can be omitted if it is `'builtins'`.

If you want to return a short name (it is enough in many times), return `__name__`.

if `__qualname__` is not available, you can use `__name__` instead.

Reply:
> if `__qualname__` is not available, you can use `__name__` instead.

This is mainly meant as replacement for `PyEval_GetFuncName` and all classes supported by that function implement `__qualname__`. So I see little reason for the additional complexity of supporting `__name__`.